src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,82 @@
+package duke.utils;
+
+import duke.commands.*;
+
+/**
+ * A utility class that parses the command inputted by the user.
+ */
+public class Parser {
+
+    /**
+     * Returns a Command object from parsing fullCommand.
+     *
+     * @param fullCommand
+     * @return A Command object containing the information from fullCommand
+     */
+    public Command parse(String fullCommand) {
+        String[] words = fullCommand.split(" ");
+        String command = fullCommand;
+        if (words.length > 1) {
+            command = words[0];
+        }
+
+        switch (command) {
+            case "bye":
+                return new ExitCommand();
+
+            case "list":
+                return new ListCommand();
+
+            case "find":
+                return new FindCommand(fullCommand);
+
+            case "delete":
+                try {
+                    int taskNumber = Integer.parseInt(words[1]);
+                    return new DeleteCommand(taskNumber);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Missing task number");
+                    Ui.printLine();
+                    return null;
+                }
```

Comment:
You can consider making a separate function to test the validity of commands, will be helpful when you want to add more user error handling.

Reply:
Okay